### ISSUE TITLE
Top-down have lower precision than bottom up

### ISSUE THREAD

**Issue by gvreergo** [open-mmlab/mmpose#736]
hi !

I'm opening an issue because I have a problem with  the fact that  every **top-down methods that have a lower precision  than bottom up** on my own dataset . While usually you should get better results with top-down-methods, I get better results on Bottom-up method. 

Here are the result of various tests I did with various datasets : https://docs.google.com/spreadsheets/d/1VwA9OIKHJP8EzJRWCUb1TzbbaCG-GnJHqsd37OPfjnQ/edit#gid=0

Every specified model used are the original config file where i just change ```data``` and ```data_cfg``` dictionnaries to match the data , annotation and bounding box file path.

I used the coco annotator tool to create my annotations. I take the json file that will be my annotation. Then make a copy of ```annotation_json["annotation"]``` that will be my the box detection result ( + a false 1.0 box confidence score i add bc it is need by the code) .
 
 ```{"id": 387, "image_id": 211, "category_id": 1, "dataset_id": 15, "segmentation": [[20.9, 256, 20.9, -0.6, 171.8, -0.6, 171.8, 256]], "area": 38656, "bbox": [21, 0, 151, 256], "iscrowd": false, "isbbox": true, "creator": "greg_is_greg", "width": 192, "height": 256, "color": "#4085ec", "keypoints": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 134, 48, 2, 89, 34, 2, 152, 130, 2, 72, 123, 2, 153, 200, 2, 51, 186, 2, 122, 192, 2, 89, 188, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], "metadata": {}, "milliseconds": 41217, "events": [{"_cls": "SessionEvent", "created_at": {"$date": 1623937046948}, "user": "greg_is_greg", "milliseconds": 22771, "tools_used": ["BBox", "Keypoints"]}, {"_cls": "SessionEvent", "created_at": {"$date": 1623938782823}, "user": "greg_is_greg", "milliseconds": 9223, "tools_used": ["BBox"]}], "num_keypoints": 8, "score": 1.0}```

here is what medium office dataset typical data looks like (with visible annotations:  
![image](https://user-images.githubusercontent.com/25591777/122798972-7c3fa900-d2c1-11eb-8d23-33f47ef49c11.png)

here is an example for medium hospital dataset : 
![image](https://user-images.githubusercontent.com/25591777/122799262-cc1e7000-d2c1-11eb-8e68-686f06ea466d.png)

Regarding the challenging images with a lot of occlusion in the hospital dataset  i kinda expected my results to be low but the fact that they are lower in TopDown detector than in BottomUp for Hospital AND Office dataset makes me think I did wrong somewhere. 

Do you have any idea of where I might have made a mistake ?

Thanks in advance for your help.

**Comment by ly015** [comment]
Hi, I think maybe you could visualize some of the top-down results, especially those with large errors. It may help to locate the problem.

**Comment by jin-s13** [comment]
What is the size of your dataset (number of images)?

**Comment by jin-s13** [comment]
If your dataset is relatively small (only a few hundred), some modifications are needed.

1. the warmup iteration should be smaller.
https://github.com/open-mmlab/mmpose/blob/202983d24665a909ae1c45f4025d66794b9e32fd/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py#L18

2. increase the number of total epochs (and accordingly increase the lr step).
https://github.com/open-mmlab/mmpose/blob/master/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py#L20-L21

3. It may help to use COCO-pretrained model to initialize the model. Replace 'None' with the model link.
https://github.com/open-mmlab/mmpose/blob/202983d24665a909ae1c45f4025d66794b9e32fd/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py#L2

**Comment by gvreergo** [comment]
@jin-s13  I used pretrained models (dowloaded from mmpose read the docs) without further training on a VERY small test database, meaning 20 pictures maximum .
I specify the checkpoint file path in the command line , does it makes a difference ? 
./tools/dist_test.sh     configs/my_config/medium_office_dataset/higher_hrnet48_coco_512x512.py  checkpoints/a_tester/higher_hrnet48_coco_512x512_ae-60fedcbc_20200712.pth  1  --out benchmark_result/medium_office_dataset/higher_hrnet_w48_512x512.json


@ly015  Most of the errors come from the fact that there are a lot of artifact on the pictures, folded blankets or intubation devices  are mistaken as joints or limbs. I have no clue of how I could remove them.

The whole point of the project is to make it work with these constraints.

**Comment by gvreergo** [comment]
@ly015  Most of the errors come from the fact that there are a lot of artifact on the pictures, folded blankets or intubation devices are mistaken as joints or limbs. I have no clue of how I could remove them.

The whole point of the project is to make it work with these constraints.
